cke-tools/src/rivers/server.go

```diff
@@ -56,7 +56,8 @@ func NewServer(upstreams []string, cfg Config) *Server {
 		dialer:    dialer,
 		pool: sync.Pool{
 			New: func() interface{} {
-				return make([]byte, copyBufferSize)
+				buf := make([]byte, copyBufferSize)
+				return &buf
```

Comment:
Why did you turn it into a pointer?

Reply:
For performance?
https://qiita.com/peroxyacyl/items/5e02ddf4480ecd2ec7b1

Reply:
- SA6002 – Storing non-pointer values in sync.Pool allocates memory
  https://staticcheck.io/docs/checks#SA6002